src/api.v2/helpers/worker/generateETag.js

```diff
@@ -0,0 +1,47 @@
+const hash = require('object-hash');
+const config = require('../../../config');
+const getExperimentBackendStatus = require('../backendStatus/getExperimentBackendStatus');
+
+const createObjectHash = (object) => hash.MD5(object);
+
+const generateETag = async (
+  experimentId,
+  body,
+  extraDependencies,
+  extras,
+  cacheUniquenessKey = null,
+) => {
+  const backendStatus = await getExperimentBackendStatus(experimentId);
+  const { pipeline: { startDate: qcPipelineStartDate } } = backendStatus;
+
+  const { workerVersion } = config;
+
+  let ETagBody;
+  // They `body` key to create ETAg for gene expression is different
+  // from the others, causing the generated ETag to be different
+  if (body.name === 'GeneExpression') {
+    ETagBody = {
+      experimentId,
+      missingGenesBody: body,
```

Comment:
The only difference between the two bodies is       missingGenesBody: body and body: body. Is there a way to generalise this so you dont add the if/else?